src/app/warehouse/warehouse.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+class Product;
+
+class Warehouse
+{
+  public:
+    void increaseAmount(const std::string &name, const uint amount);
+    void decreaseAmount(const std::string &name, const uint amount);
+    void addProducts(const std::vector<std::shared_ptr<Product>> &products);
+    bool validateAdd(const std::string &name) const;
+    void removeProducts(const std::vector<std::string> &products);
+    bool validateRemoval(const std::string &name) const;
+    std::shared_ptr<Product> getPtrToProduct(const std::string &name);
+    std::vector<std::shared_ptr<Product>> &getExpiredProducts(); // TODO: write this function when type Date will be
+                                                                 // created
+    std::vector<std::shared_ptr<Product>> &getProducts();
+
+  private:
+    bool is_valid_state;
+    std::vector<std::shared_ptr<Product>> products_;
+};
```

Comment:
Change class structure so private members are at the top of the class.
Add class extent.

Reply:
Why class extent is needed here?

Reply:
We want to make class extent in every class, no matter how many instances gonna be initialised. Right now we haven't made a decision if we'll have only one or few warehouses, no matter the count, we should always have static function that points to an instance of an object.

Reply:
Done